.cbmc-batch/stubs/evp_md_ctx_is_valid_no_pkey_stub.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <ec_utils.h>
+#include <make_common_data_structures.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include <openssl/kdf.h>
+#include <openssl/rsa.h>
+#include <proof_helpers/nondet.h>
+
+/* Helper function for CBMC proofs: checks if EVP_MD_CTX is valid.
+ * Use this stub when we are *certain* there is no pkey associated with the digest context.
+ */
+bool evp_md_ctx_is_valid(EVP_MD_CTX *ctx) {
+    return ctx && ctx->is_initialized && ctx->digest_size <= EVP_MAX_MD_SIZE && ctx->pkey == NULL;
```

Comment:
If we want to be *certain* to only use this stub when there is no `pkey` associated with the digest context, shouldn't we assert that here?
```c
assert(ctx->pkey == NULL);
```

Reply:
If that isn't the case and we really want to this be a part of the conjunction, update the documentation.

Reply:
Yes, I think the assert is better and ensures that we really only use this stub when there is no pkey. I have updated this!